Paintroid/src/main/java/org/catrobat/paintroid/dialog/LayersDialog.java

```diff
@@ -0,0 +1,483 @@
+/**
+ *  Paintroid: An image manipulation application for Android.
+ *  Copyright (C) 2010-2015 The Catrobat Team
+ *  (<http://developer.catrobat.org/credits>)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.paintroid.dialog;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.graphics.Bitmap;
+import android.graphics.Color;
+import android.os.Bundle;
+import android.text.InputFilter;
+import android.text.InputType;
+import android.util.Log;
+import android.view.View;
+import android.view.WindowManager;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.AdapterView;
+import android.widget.AdapterView.OnItemClickListener;
+import android.widget.AdapterView.OnItemLongClickListener;
+import android.widget.EditText;
+import android.widget.GridView;
+import android.widget.ImageButton;
+import android.widget.SeekBar;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import org.catrobat.paintroid.MainActivity;
+import org.catrobat.paintroid.PaintroidApplication;
+import org.catrobat.paintroid.R;
+import org.catrobat.paintroid.command.implementation.LayerCommand;
+import org.catrobat.paintroid.eventlistener.ChangeActiveLayerEventListener;
+import org.catrobat.paintroid.eventlistener.RefreshLayerDialogEventListener;
+import org.catrobat.paintroid.tools.Layer;
+import org.catrobat.paintroid.ui.button.LayersAdapter;
+
+import java.util.ArrayList;
+
+public final class LayersDialog extends BaseDialog implements OnItemClickListener,
+	OnItemLongClickListener, DialogInterface.OnDismissListener,
+		SeekBar.OnSeekBarChangeListener, RefreshLayerDialogEventListener, ChangeActiveLayerEventListener {
+
+	private static final String NOT_INITIALIZED_ERROR_MESSAGE = "LayerDialog has not been initialized. Call init() first!";
+
+	private static LayersDialog instance;
+
+	private LayersAdapter mLayerButtonAdapter;
```

Comment:
regarding intentation, pls always use tabulators and not spaces. Github has problems with correct intentation if spaces are used instead of tabs